extensions/control-plane/api/management-api/contract-definition-api/src/test/java/org/eclipse/edc/connector/api/management/contractdefinition/ContractDefinitionApiControllerIntegrationTest.java

```diff
@@ -220,15 +221,55 @@ void deleteContractDefinition_notExists() {
                 .statusCode(404);
     }
 
-    private ContractDefinitionRequestDto createDto(String definitionId) {
-        return ContractDefinitionRequestDto.Builder.newInstance()
+    @Test
+    void updateContractDefinition_whenExists(ContractDefinitionStore store) {
+
+        store.accept(createContractDefinition("definitionId"));
+
+        var dto = updateDto();
```

Comment:
With the current test setup it's possible that the test is green, but no actual update has happened.
I'd suggest using the original DTO, then updating it (e.g. the `AssetSelectorExpression`), and then asserting that the updated DTO is indeed in the `store`.